### ISSUE TITLE
Allow for different Inferencing regimes with inference arg

### ISSUE THREAD

**Issue by nicholascar** [RDFLib/pySHACL#2]
Instead of the inference arg being a bool, it should be a string for 'none', 'owl-rl', 'rdfs' or 'both' with default being 'both' - the various OWL-RL inference regimes or none.

**Comment by ashleysommer** [comment]
This was already changed in commit 
https://github.com/RDFLib/pySHACL/commit/6dcbd5722ce3e89f8c0c9e44191e0730bf4fabce to be the way you described.

I've run tests with rdfs, owl-rl, and both, and found that for all of the tests from the standard core SHACL test suite files, all pass with just rdfs inferencing, and in fact some OWL-RL inferencing atrtifacts mess up validator results vs the the expected results hardcoded in the test SHACL files.

Another point to consider, the validator takes approx 10x-15x longer to run on a OWL-RL inferenced targer graph vs a RDFS inferenced target graph.

For those two reasons I've chosen 'rdfs' to be the default inferencing option.

**Comment by nicholsn** [comment]
Out of curiosity, what is the performance difference comparing `RDFS` to `None`?

Generally, I would think you would want to go with `None` as the default, rather than adding additional triples into the graph. 

At least I would find that behavior more intuitive, and if I want to enable additional features, then that is my choice/intention. But as long as it is clearly documented and the options are present to disable inferencing, then we are all good.

**Comment by ashleysommer** [comment]
I've added a benchmark.py script using the built-in python `timeit` module to do execution time measurement for each pre-inferencing method:

Here's what I've found:
1) Running in python Debug mode:
```
Benchmark completed. Validation took:
With no inferencing: 0.0162 seconds
With rdfs inferencing: 0.1638 seconds
With owl-rl inferencing: 0.4399 seconds
With both inferencing: 2.4329 seconds
```
2) Running in python normal (non-debug) mode:
```
Benchmark completed. Validation took:
With no inferencing: 0.0065 seconds
With rdfs inferencing: 0.0639 seconds
With owl-rl inferencing: 0.1685 seconds
With both inferencing: 0.8945 seconds
```

These numbers are illustrative of the magnitude of difference in time taken to run a SHACL validation using each of the pre-inferencing options.

I realise now heaver that quite a major contributing factor of the time difference seen is caused by the cost of running the inferencing step itself (which _is_ included in the time taken measured in the benchmark) as opposed to the added overhead of a running the validator on a graph with heavier inferenced graph as I originally thought was the cause.

My main takeaway from this is looking at the time taken with 'none' inferencing. At just 6.5 milliseconds from start to finish, including generating and printing the validation report, this validator is _surprisingly_  fast, considering it is a pure-python implemention.
